--- postgis/long_xact.c
+++ postgis/long_xact.c
@@ -3,6 +3,7 @@
 #include "executor/spi.h"       /* this is what you need to work with SPI */
 #include "commands/trigger.h"   /* ... and triggers */
 #include "utils/lsyscache.h"	/* for get_namespace_name() */
+#include "utils/rel.h"

 #include "lwgeom_pg.h"

--- postgis/lwgeom_estimate.c
+++ postgis/lwgeom_estimate.c
@@ -19,6 +19,7 @@
 #include "utils/array.h"
 #include "utils/lsyscache.h"
 #include "utils/syscache.h"
+#include "utils/rel.h"

 #include "liblwgeom.h"
 #include "lwgeom_pg.h"
--- postgis/lwgeom_triggers.c
+++ postgis/lwgeom_triggers.c
@@ -1,6 +1,7 @@
 #include "postgres.h"
 #include "executor/spi.h"       /* this is what you need to work with SPI */
 #include "commands/trigger.h"   /* ... and triggers */
+#include "utils/rel.h"
 #include "lwgeom_pg.h"

 Datum cache_bbox(PG_FUNCTION_ARGS);
